const loadScript = async (Src) => {
    return new promise ((resolve,reject)=>{
      let script = dcument.creatElement("script")
      script.src = src
      script.onload = () =>{
        resolve(src + "done success")
         }
      document.head.append(script)
    })
  }
  // Problem number 1
  // let a = loadScript("https://cdn.jsdelivr.net/npm/bootstrap@5.2.2/dist/js/bootstrap.bundle.min.js")
  // a.then((value) => {
  //         console.log(value)
  // })
  
  // Problem number 2
  // const main2 = async () => {
  //         console.log(new Date().getMilliseconds())
  //         let a = await loadScript("https://cdn.jsdelivr.net/npm/bootstrap@5.2.2/dist/js/bootstrap.bundle.min.js")
  //         console.log(a)
  //         console.log(new Date().getMilliseconds())
  // }
  // main2()
  
  // Problem number 3
  // let p = () => {
  //         return new Promise((resolve, reject) => {
  //                 setTimeout(() => {
  //                         reject(new Error("Please this is not acceptable"))
  //                 }, 3000)
  //         })
  // }
  
  // let a = async () => {
  //         try {
  //                 let c = await p()
  //                 console.log(c)
  //         }
  //         catch (err) {
  //                 console.log("This error has been handled")
  //         }
  // }
  // a()
  
  // Problem number 4
  
  let p1 = async () => {
          return new Promise((resolve, reject) => {
                  setTimeout(() => {
                          resolve(10)
                  }, 2000)
          })
  }
  let p2 = async () => {
          return new Promise((resolve, reject) => {
                  setTimeout(() => {
                          resolve(20)
                  }, 1000)
          })